include/verilated.h

```diff
@@ -2487,6 +2488,44 @@ static inline void VL_ASSIGNSEL_WIIW(int obits, int lsb, WDataOutP owp, WDataInP
     _VL_INSERT_WW(obits, owp, rwp, lsb + obits - 1, lsb);
 }
 
+//======================================================================
+// Variable randomization (std::randomize, randomize method)
+
+#define VL_DEF_RANDOMIZE_VAR(varType) \
+inline IData VL_RANDOMIZE_VAR(varType& odata, int obits) VL_MT_SAFE { \
+    odata = VL_RANDOM_Q(obits); \
+    return 1; \
+}
+VL_DEF_RANDOMIZE_VAR(CData)
+VL_DEF_RANDOMIZE_VAR(SData)
+VL_DEF_RANDOMIZE_VAR(IData)
+VL_DEF_RANDOMIZE_VAR(QData)
+
+#include <iostream>
+
+inline IData VL_RANDOMIZE_VAR(WDataOutP odata, int lsb, int obits) VL_MT_SAFE {
+    VL_ASSIGNSEL_WIIQ(obits, lsb, odata, VL_RANDOM_Q(obits));
+    return 1;
+}
+
+#define VL_DEF_RANDOMIZE_ENUM_VAR(varType) \
+template<size_t valCnt> \
+inline IData VL_RANDOMIZE_ENUM_VAR(varType& odata, const std::array<varType, valCnt>& values) VL_MT_SAFE { \
+    odata = values[vl_rand64() % values.size()]; \
+    return 1; \
+}
+
+VL_DEF_RANDOMIZE_ENUM_VAR(CData)
+VL_DEF_RANDOMIZE_ENUM_VAR(SData)
+VL_DEF_RANDOMIZE_ENUM_VAR(IData)
+VL_DEF_RANDOMIZE_ENUM_VAR(QData)
+
+template<size_t valCnt>
+inline IData VL_RANDOMIZE_ENUM_VAR(WDataOutP odata, const std::array<QData, valCnt>& values, int lsb, int obits) VL_MT_SAFE {
+    VL_ASSIGNSEL_WIIQ(obits, lsb, odata, values[vl_rand64() % values.size()]);
```

Comment:
This would result in a lot of code as the template will replicate the function, think not needed if do above.